### ISSUE TITLE
SelectOneMenu: Missing aria-disabled attribute when disabled="true"

### ISSUE THREAD

**Issue by philipmathen-cc** [primefaces/primefaces#14625]
### Describe the bug

The **<p:selectOneMenu/>** component does not set the `aria-disabled` attribute when `disabled="true"`. When using a screen-reader, the disabled state of the element is therefore not communicated to the user. This is an accessibility issue. 

It was tested with Windows Narrator (Windows 11) and NVDA 2021.2 as well.

### Reproducer

1. Create  <p:selectOneMenu disabled="true"/> 
2. Visually the element looks disabled
3. Inspect the rendered element in the browser -> no `aria-disabled="true"` attribute gets rendered
4. Start narrator on windows `(ctrl + windows + enter)` and press `tab` until you reach the element
5. Element gets read like a non-disabled element

### Code analysis
The javadoc for renderAccessibilityAttributes() in InputRenderer.java explicitly states, that the `aria-disabled` attribute will be rendered, which the code is currently missing.
```java
/**
 * Adds the following accessibility attributes to an HTML DOM element.
 * <pre>
 * "aria-required" if the component is required
 * "aria-invalid" if the component is invalid
 * "aria-labelledby" if the component has a labelledby attribute
 * "disabled" and "aria-disabled" if the component is disabled
 * "readonly" and "aria-readonly" if the component is readonly
 * </pre>
 * @param context the {@link FacesContext}
 * @param component the {@link UIInput} component to add attributes for
 * @throws IOException if any error occurs writing the response
 */
protected void renderAccessibilityAttributes(FacesContext context, UIInput component) throws IOException {
    renderAccessibilityAttributes(context, component, isDisabled(component), isReadOnly(component));
}

protected void renderAccessibilityAttributes(FacesContext context, UIInput component, boolean disabled, boolean readonly)
                throws IOException {
        ResponseWriter writer = context.getResponseWriter();

        renderARIARequired(context, component);
        renderARIAInvalid(context, component);

        if (component instanceof InputHolder) {
            InputHolder inputHolder = ((InputHolder) component);
            String labelledBy = inputHolder.getLabelledBy();
            if (LangUtils.isNotBlank(labelledBy)) {
                writer.writeAttribute(HTML.ARIA_LABELLEDBY, labelledBy, null);
            }
            String ariaDescribedBy = inputHolder.getAriaDescribedBy();
            if (LangUtils.isNotBlank(ariaDescribedBy)) {
                String target = SearchExpressionUtils.resolveClientIds(ariaDescribedBy, component);
                writer.writeAttribute(HTML.ARIA_DESCRIBEDBY, target, null);
            }
        }

        String ariaDescribedBy = null;
        if (component instanceof InputAware) {
            ariaDescribedBy = ((InputAware) component).getAriaDescribedBy();
            if (LangUtils.isNotBlank(ariaDescribedBy)) {
                String target = SearchExpressionUtils.resolveClientIds(ariaDescribedBy, component);
                writer.writeAttribute(HTML.ARIA_DESCRIBEDBY, target, null);
            }
        }

        if (disabled) {
            writer.writeAttribute("disabled", "disabled", null);
            // The following line is missing
            // writer.writeAttribute(HTML.ARIA_DISABLED, "true", null);

        }

        if (readonly) {
            writer.writeAttribute("readonly", "readonly", null);
        }
    }
````

### Expected behavior

Setting `disabled="true"` on <p:selectOneMenu/> should lead to `aria-disabled="true"` in the rendered component. 

### PrimeFaces edition

None

### PrimeFaces version

15.0.6

### JSF implementation

Mojarra

### JSF version

4.0

### Java version

17

**Comment by melloware** [comment]
@philipmathen-cc it used to be there but i removed it with this ticket and PR: https://github.com/primefaces/primefaces/issues/8470

PR: https://github.com/primefaces/primefaces/pull/8471

**Comment by melloware** [comment]
Looks like its a W3C violation of the ARIA spec

**Comment by melloware** [comment]
i think this issue is different

```xml
<div id="j_id_47:option" class="ui-selectonemenu ui-widget ui-state-default ui-state-focus" aria-owns="j_id_47:option_items">
	<div class="ui-helper-hidden-accessible">
		<select id="j_id_47:option_input" name="j_id_47:option_input" tabindex="-1" autocomplete="off" aria-hidden="true" aria-label="Basic" data-p-label="Basic" data-p-hl="onemenu">
			<option value="" data-escape="true">Select One</option>
			<option value="Option1" data-escape="true">Option1</option>
			<option value="Option2" data-escape="true">Option2</option>
			<option value="Option3" data-escape="true">Option3</option>
		</select>
	</div>
	<span id="j_id_47:option_label" class="ui-selectonemenu-label ui-inputfield" tabindex="0" aria-controls="j_id_47:option_panel" aria-label="Select One" role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-labelledby="j_id_47:j_id_49" aria-activedescendant="j_id_47:option_0" aria-disabled="false">Select One</span>
	<div class="ui-selectonemenu-trigger ui-state-default ui-state-focus">
		<span class="ui-icon ui-icon-triangle-1-s ui-c"/>
	</div>
</div>
```

that is the rendered HTML i think the screenreader is reading the `span id="j_id_47:option_label" class="ui-selectonemenu-label ` with `aria-label="Select One"` that is getting read

**Comment by melloware** [comment]
here is disabled:

```xml
<div id="j_idt143:disabled" class="ui-selectonemenu ui-widget ui-state-default ui-state-disabled">
	<div class="ui-helper-hidden-accessible">
		<select id="j_idt143:disabled_input" name="j_idt143:disabled_input" tabindex="-1" autocomplete="off" aria-hidden="true" aria-label="Disabled" disabled="disabled" data-p-label="Disabled" data-p-hl="onemenu">
			<option value="" data-escape="true">Select One</option>
			<option value="Option1" data-escape="true">Option1</option>
			<option value="Option2" data-escape="true">Option2</option>
			<option value="Option3" data-escape="true">Option3</option>
		</select>
	</div>
	<span id="j_idt143:disabled_label" class="ui-selectonemenu-label ui-inputfield" tabindex="0" aria-controls="j_idt143:disabled_panel" aria-label="Select One" role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-labelledby="j_idt143:j_idt210" disabled="disabled">Select One</span>
	<div class="ui-selectonemenu-trigger ui-state-default">
		<span class="ui-icon ui-icon-triangle-1-s ui-c"/>
	</div>
	<div id="j_idt143:disabled_panel" class="ui-selectonemenu-panel ui-widget ui-widget-content ui-helper-hidden ui-shadow ui-input-overlay" tabindex="-1">
		<div class="ui-selectonemenu-items-wrapper" style="max-height:200px"/>
	</div>
</div>
```

**Comment by philipmathen-cc** [comment]
You´re right, the span gets read by the screenreader. The problem is that a `<span/>` does not support the disabled attribute.

>The disabled attribute is supported by `<button>`, `<fieldset>`, `<optgroup>`, `<option>`, `<select>`, `<textarea>` and `<input>`.

[HTML attribute: disabled](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/disabled)


In order for the screenreader to acknowledge the disabled state `aria-disabled` needs to be set on the `<span/>`. 
>Another reason to use the aria-disabled attribute over the HTML disabled attribute is if you have created custom controls which need to be marked as disabled, but are not using an element that allows for the disabled attribute. 

[ARIA: aria-disabled attribute](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-disabled)

**Comment by melloware** [comment]
Yep thanks! I will fix for 15.0.13

**Comment by melloware** [comment]
After my fix. `<span>` no longer has `disabled` but has `aria-disabled`

```xml
<div id="j_idt143:disabled" class="ui-selectonemenu ui-widget ui-state-default ui-state-disabled">
	<div class="ui-helper-hidden-accessible">
		<select id="j_idt143:disabled_input" name="j_idt143:disabled_input" tabindex="-1" autocomplete="off" aria-hidden="true" aria-label="Disabled" disabled="disabled" data-p-label="Disabled" data-p-hl="onemenu">
			<option value="" data-escape="true">Select One</option>
			<option value="Option1" data-escape="true">Option1</option>
			<option value="Option2" data-escape="true">Option2</option>
			<option value="Option3" data-escape="true">Option3</option>
		</select>
	</div>
	<span id="j_idt143:disabled_label" class="ui-selectonemenu-label ui-inputfield" tabindex="0" aria-controls="j_idt143:disabled_panel" aria-disabled="true" aria-label="Select One" role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-labelledby="j_idt143:j_idt210">Select One</span>
	<div class="ui-selectonemenu-trigger ui-state-default">
		<span class="ui-icon ui-icon-triangle-1-s ui-c"/>
	</div>
	<div id="j_idt143:disabled_panel" class="ui-selectonemenu-panel ui-widget ui-widget-content ui-helper-hidden ui-shadow ui-input-overlay" tabindex="-1">
		<div class="ui-selectonemenu-items-wrapper" style="max-height:200px"/>
	</div>
</div>
```

**Comment by Joern2026** [comment]
Hi @melloware,
for me it seems that the labelFor behavior is broken with your fix in SelectOneMenuRenderer. In case of non editable SelectOneMenu's renderAccessibilityAttributes is not invoked anymore and the aria-labelledby not generated. This is an issue in our application with PF 15.0.13. Can you confirm that?

best regards

**Comment by melloware** [comment]
@Joern2026 are you saying yours is only an issue with `editable="false"` but its fine if `editable="true"` ?

**Comment by Joern2026** [comment]
@melloware In our application, we have a case where the SelectOneMenu is not "editable" and has a label (p:outputLabel with a for attribute). With version 15.0.13, 'aria-labelledby' is no longer generated. I think because of your change in Line 299.

**Comment by melloware** [comment]
Understood. let me see if I can fix for 15.0.14

**Comment by tandraschko** [comment]
please also add some integration tests to verify the correct behavior

**Comment by melloware** [comment]
integration test added

**Comment by Joern2026** [comment]
@melloware Thank you very much for your adjustments.

**Comment by melloware** [comment]
15.0.14 i am publishing now should be in Maven Central in the new couple of hours
